src/contexts/providers.context.tsx

```diff
@@ -60,45 +77,79 @@ const ProvidersProvider: FC<PropsWithChildren> = (props) => {
     }
   };
 
+  interface ConnectMetamaskProviderParams {
+    env: Env;
+    web3Provider: Web3Provider;
+    account: string;
+  }
+
+  const connectMetamaskProvider = useCallback(
+    async ({ env, web3Provider, account }: ConnectMetamaskProviderParams): Promise<void> => {
+      // ToDo: Why is this try/catch required in a promise?
+      try {
+        const checkMetamaskHeartbeat = setTimeout(() => {
+          setAccount({
+            status: "failed",
+            error: `It seems that ${WalletName.METAMASK} is not responding to our requests\nPlease reload the page and try again`,
+          });
+        }, 3000);
+
+        const currentNetwork = await web3Provider.getNetwork();
+        clearTimeout(checkMetamaskHeartbeat);
+
+        const currentNetworkChainId = currentNetwork.chainId;
+        const supportedChainIds = env.chains.map((chain) => chain.chainId);
+        if (!supportedChainIds.includes(currentNetworkChainId)) {
+          setAccount({
+            status: "failed",
+            error: supportedChainIds.includes(EthereumChainId.GOERLI)
+              ? `Switch your network to Ethereum Goerli testnet or ${env.chains[1].name} to continue`
+              : `Switch your network to Ethereum or ${env.chains[1].name} to continue`,
+          });
+        } else {
+          setAccount({ status: "successful", data: account });
+          setConnectedProvider({
+            status: "successful",
+            data: { provider: web3Provider, chainId: currentNetworkChainId },
+          });
+          return setAccount({ status: "successful", data: account });
```

Comment:
Yo are setting the account twice here and in L110.

Reply:
Good catch! Fixed.